triton-cli/src/main.rs

```diff
@@ -89,171 +89,122 @@ fn parse_inputs(inputs: Option<String>) -> Vec<BFieldElement> {
         .unwrap_or_default()
         .split(',')
         .filter(|v| !v.is_empty())
-        .map(|v| BFieldElement::from(v.parse::<u64>().unwrap()))
-        .collect::<Vec<BFieldElement>>()
+        .map(|v| v.parse().unwrap())
+        .collect()
 }
 
 fn prove(
-    asm: &String,
-    out: &String,
+    asm_filepath: &str,
+    out: &str,
     public_inputs: Option<String>,
     private_inputs: Option<String>,
-) {
-    // TODO: test paths before making the proof?
-    let asm = fs::read_to_string(asm).expect("unable to read file");
-    let instructions = triton_vm::parser::parse(&asm).unwrap();
-    let l_instructions = triton_vm::parser::to_labelled_instructions(instructions.as_slice());
-    let program = triton_vm::program::Program::new(l_instructions.as_slice());
+) -> Result<()> {
+    if std::path::Path::new(out).exists() {
+        println!("output file already exists: {out}");
+        std::process::exit(1);
+    }
+    let asm = fs::read_to_string(asm_filepath)
+        .with_context(|| "Failed to read Triton assembly from file")?;
+    let program = triton_program!({ asm });
 
     let public_input = PublicInput::from(parse_inputs(public_inputs));
     let non_determinism = NonDeterminism::from(parse_inputs(private_inputs));
     println!("proving...");
-    let data = triton_vm::prove_program(&program, public_input, non_determinism).unwrap();
+    let data = triton_vm::prove_program(&program, public_input, non_determinism)
+        .with_context(|| "Triton VM errored during program execution")?;
     println!("success!");
 
-    // write the proof in an arbitrary binary format
-    //
-    // version: u8
-    //
-    // security_level: u64
-    // fri_expansion_factor: u64
-    // num_trace_randomizers: u64
-    // num_colinearity_checks: u64
-    // num_combination_codeword_checks: u64
-    //
-    // digest_length: u8
-    // public_input_length: u64
-    // public_output_length: u64
-    // proof_length: u64
-    //
-    // program_digest: u64[digest_length]
-    //
-    // public_inputs: u64[public_input_length]
-    // public_outputs: u64[public_output_length]
-    // proof: u64[proof_length]
-    write_proof(data, out).expect("failed to write proof to file");
+    write_proof(data, out).with_context(|| "Failed to write proof to file")?;
     println!("proof written to: {out}");
+    Ok(())
+}
+
+#[derive(Serialize, Deserialize)]
+struct SerializedProof {
+    version: u8,
+    security_level: u64,
+    fri_expansion_factor: u64,
+    num_trace_randomizers: u64,
+    num_colinearity_checks: u64,
+    num_combination_codeword_checks: u64,
+    program_digest: Vec<u64>,
+    public_inputs: Vec<u64>,
+    public_outputs: Vec<u64>,
+    proof: Vec<u64>,
 }
 
-fn read_proof(proof_path: &String) -> Result<(Stark, Claim, Proof)> {
-    let file = fs::File::open(proof_path)?;
-    let mut reader = BufReader::new(file);
-    let version = read_u8(&mut reader);
-    assert!(version == 1, "wrong proof file version!");
```

Comment:
Was the deletion of this line accidental, or do you simply accept any version specificer for now, anticipating restrictions in the future?

Reply:
accidental, good catch